src/ManifestStrategy/GlossaryManifestStrategy.php

```diff
@@ -0,0 +1,166 @@
+     * @return array<array-key, string>
+     */
+    protected function getGlossaryFileLinesForAdd(
+        array $manifest,
+        array $existingGlossaryFileLines
+    ): array {
+        $indexGlossaryLinesByGlossaryKeysAndLanguages = $this->indexGlossaryLinesByGlossaryKeysAndLanguages(
+            $existingGlossaryFileLines,
+        );
+        $glossaryFileLinesForAdd = [];
+        foreach ($manifest as $glossaryKey => $glossaryValues) {
+            $glossaryKeyFileLinesForAdd = $this->getGlossaryKeyFileLinesFromGlossaryKey(
+                $glossaryKey,
+                $glossaryValues,
+                $indexGlossaryLinesByGlossaryKeysAndLanguages,
+            );
+            $glossaryFileLinesForAdd = array_merge($glossaryFileLinesForAdd, $glossaryKeyFileLinesForAdd);
+        }
+
+        return $glossaryFileLinesForAdd;
+    }
+
+    /**
+     * @param string $glossaryKey
+     * @param array<string, string> $glossaryValues
+     * @param array<string, array<string, string>> $indexGlossaryLinesByGlossaryKeysAndLanguages
+     *
+     * @return array<array-key, string>
+     */
+    protected function getGlossaryKeyFileLinesFromGlossaryKey(
+        string $glossaryKey,
+        array $glossaryValues,
+        array $indexGlossaryLinesByGlossaryKeysAndLanguages
+    ): array {
+        $glossaryKeyFileLines = [];
+        foreach ($glossaryValues as $glossaryKeyLanguage => $glossaryKeyValue) {
+            if (isset($indexGlossaryLinesByGlossaryKeysAndLanguages[$glossaryKey][$glossaryKeyLanguage])) {
+                continue;
+            }
+            $glossaryKeyFileLines[] = $this->createGlossaryFileLine(
+                $glossaryKey,
+                $glossaryKeyLanguage,
+                $glossaryKeyValue,
+            );
+        }
+
+        return $glossaryKeyFileLines;
+    }
+
+    /**
+     * @param string $glossaryKey
+     * @param string $glossaryKeyLanguage
+     * @param string $glossaryKeyValue
+     *
+     * @return string
+     */
+    protected function createGlossaryFileLine(string $glossaryKey, string $glossaryKeyLanguage, string $glossaryKeyValue): string
+    {
+        return implode(',', [
+            $glossaryKey,
+            $glossaryKeyValue,
+            $glossaryKeyLanguage,
+        ]);
+    }
+
+    /**
+     * @param array<array-key, string> $glossaryExistingFileLines
+     *
+     * @return array<string, array<string, string>>
+     */
+    protected function indexGlossaryLinesByGlossaryKeysAndLanguages(array $glossaryExistingFileLines): array
+    {
+        $indexGlossaryLinesByGlossaryKeysAndLanguages = [];
+        foreach ($glossaryExistingFileLines as $glossaryLine) {
+            $glossaryLineParts = explode(',', $glossaryLine);
+            if (count($glossaryLineParts) !== static::GLOSSARY_LINE_PARTS_COUNT) {
+                continue;
+            }
+            $indexGlossaryLinesByGlossaryKeysAndLanguages[$glossaryLineParts[0]][$glossaryLineParts[2]] = trim($glossaryLine);
```

Comment:
mb we should put value here? I see that you are using it just to check if such record exists, but it doesn't make sense to keep the full line as a value here.

It should be rather some boolean value just to show that such records exists, or it should be actual translation value that we can use for some other case.

Reply:
Also, personally I would prefer to have keys as constants with meaningful names.
E.g. GLOSSARY_INDEX_KEY and GLOSSARY_INDEX_LANGUAGE